hasher/src/roinput.rs

```diff
@@ -98,10 +98,8 @@ impl ROInput {
 
     /// Append a scalar field element
     pub fn append_scalar(mut self, s: Fq) -> Self {
```

Comment:
Changing the implementation of this method implies introducing a breaking changes in the legacy signature, and in particular it implies we must change the test vectors for the signature algorithm in 3c6c4aa5e0759cf37a00a520037db4a01e320e24.
This is not a behavior we would like to introduce. Can you create a separate method instead please?